ai/default/aiunit.cpp

```diff
@@ -641,7 +641,8 @@ static void dai_military_bodyguard(struct ai_type *ait,
       int unit_mv_rate = unit_move_rate(punit);
       fc_assert_ret_msg(unit_mv_rate, "div by zero");
       if (me2goal < me2them
-          || (me2goal / unit_mv_rate < them2goal / unit_move_rate(aunit)
+          || (me2goal / unit_mv_rate
+                  < them2goal / std::min(1, unit_move_rate(aunit))
               && me2goal / unit_mv_rate < me2them / unit_mv_rate
               && unit_mv_rate > unit_move_rate(aunit))) {
```

Comment:
We have two techniques to protect against zero `unit_move_rate` now... Can we keep using the same?
```cpp
      int punit_mv_rate = unit_move_rate(punit);
      int aunit_mv_rate = unit_move_rate(aunit);
      fc_assert_ret_msg(punit_mv_rate, "div by zero");
      fc_assert_ret_msg(aunit_mv_rate, "div by zero");
      if (me2goal < me2them
          || (me2goal / punit_mv_rate < them2goal / aunit_move_rate
              && me2goal / unit_mv_rate < me2them / punit_mv_rate
              && unit_mv_rate > aunit_move_rate)) {
```